### DISCUSSION TITLE
dependency:  curl with ssl support needs libssl

### DISCUSSION THREAD

**Discussion by fraternl** [Freetz-NG/freetz-ng#900] [Allgemein]
on current master, curl with ssl support doesn't automatically select libssl
It then throws this error:
```
curl ifconfig.co
/usr/bin/curl: can't load library 'libssl.so.3'

```

I changed Config.in, so it automatically select libssl when you select "SSL Support" in curl.
It was working before because I had another package that needed libssl.
I'm not using that package anymore and I did a fresh clone.

diff make/pkgs/curl/Config.in make/pkgs/curl/Config.in.org
```
19d18
<               select FREETZ_LIB_libssl

```
![image](https://github.com/Freetz-NG/freetz-ng/assets/8692856/5effcb0c-2397-496f-8245-2bf37ed473cc)

**Comment by fda77** [comment]
Is this only for openssl3 or also the older versions?

**Comment by fraternl** [comment]
I didn't select the deprecated versions and now it is working, so I guess it's only openssl3 that needs to be selected.

![image](https://github.com/Freetz-NG/freetz-ng/assets/8692856/b80dc54e-c99c-4266-a5f3-670add8b42f8)
![image](https://github.com/Freetz-NG/freetz-ng/assets/8692856/9593a79e-4856-4343-aecf-429c788986c6)

```
curl ifconfig.co
142.250.179.142

```

**Comment by fda77** [comment]
So lets wait until someone checked all 4 openssl versions in freetz to know if curl need libssl always now or its something new in openssl3
